aem-mock/core/src/main/java/io/wcm/testing/mock/aem/MockTag.java

```diff
@@ -316,19 +317,31 @@ public boolean isNamespace() {
   public String getXPathSearchExpression(String property) {
     String tagRoot = MockTagManager.getTagRootPath();
     String ns = getNamespace().getName();
-    String relPath = StringUtils.substringAfter(getPath(), tagRoot + "/" + ns + "/");
+    String relPath = StringUtils.substringAfter(getPath(), tagRoot + '/' + ns + '/');
     boolean isDefaultNamespace = StringUtils.equals(ns, TagConstants.DEFAULT_NAMESPACE);
+
+    final String propertyName = StringUtils.removeStart(StringUtils.contains(property, String.valueOf('/'))
```

Comment:
probably the code and the intention would be clearer with parsing the property string using a regex? otherwise put in a comment what the two lines are actually doing and why.

Reply:
I'll refactor it a little bit and put some comments to make it a bit clearer.

It would have been so much easier if there was an inverted "substringBeforeLast" and "substringAfterLast" (that would return the opposite value if the needle wasn't found it in the haystack, instead of an empty string.